batch/python/resample.py

```diff
@@ -0,0 +1,539 @@
+
+
+def world_bounds(crs: CRS) -> Bounds:
+    """Get the world bounds for a given CRS.
+
+    Taken from pixetl
+    """
+
+    from_crs = CRS(4326)
+
+    proj = Transformer.from_crs(from_crs, crs, always_xy=True)
+
+    _left, _bottom, _right, _top = crs.area_of_use.bounds
+
+    # Get World Extent in Source Projection
+    # Important: We have to get each top, left, right, bottom separately.
+    # We cannot get them using the corner coordinates.
+    # For some projections such as Goode (epsg:54052) this would cause strange behavior
+    top = proj.transform(0, _top)[1]
+    left = proj.transform(_left, 0)[0]
+    bottom = proj.transform(0, _bottom)[1]
+    right = proj.transform(_right, 0)[0]
+
+    return left, bottom, right, top
+
+
+def reproject_bounds(bounds: Bounds, src_crs: CRS, crs: CRS) -> Bounds:
+    """Reproject source bounds from source CRS to destination CRS.
+
+    Make sure that coordinates fall within real world coordinates system
+    Taken from pixetl
+    """
+
+    left, bottom, right, top = bounds
+
+    min_lng, min_lat, max_lng, max_lat = world_bounds(crs)
+
+    proj = Transformer.from_crs(CRS.from_user_input(src_crs), crs, always_xy=True)
+
+    reproject_top = replace_inf_nan(round(proj.transform(0, top)[1], 8), max_lat)
+    reproject_left = replace_inf_nan(round(proj.transform(left, 0)[0], 8), min_lng)
+    reproject_bottom = replace_inf_nan(round(proj.transform(0, bottom)[1], 8), min_lat)
+    reproject_right = replace_inf_nan(round(proj.transform(right, 0)[0], 8), max_lng)
+
+    return reproject_left, reproject_bottom, reproject_right, reproject_top
+
+
+def create_vrt(
+    uris: List[str],
+    src_file_band: Optional[int] = None,
+    vrt_path: str = "all.vrt",
+    separate=False,
+) -> str:
+    """Create a VRT file from input URI(s)
+
+    Adapted from pixetl.
+    """
+    input_file_list_string = "\n".join(uris)
+
+    with tempfile.TemporaryDirectory() as temp_dir_name:
+        input_list_file_name = os.path.join(temp_dir_name, "input_file_list.txt")
+
+        with open(input_list_file_name, "w") as input_file_list_file:
+            input_file_list_file.write(input_file_list_string)
+
+        cmd: List[str] = ["gdalbuildvrt"]
+
+        cmd += ["-input_file_list", input_list_file_name]
+
+        if src_file_band is not None:
+            cmd += ["-b", str(src_file_band)]
+        if separate:
+            cmd += ["-separate"]
+        cmd += ["-resolution", "highest"]
+        cmd += [vrt_path]
+
+        vrt_process = subprocess.run(cmd, capture_output=True)
+        if vrt_process.returncode < 0:
+            raise SubprocessKilledError
+        if vrt_process.returncode > 0:
+            # TODO: Log output
+            raise Exception("Error creating VRT!")
+
+    return vrt_path
+
+
+def get_s3_client(aws_region=AWS_REGION, endpoint_url=AWS_ENDPOINT_URL):
+    return boto3.client("s3", region_name=aws_region, endpoint_url=endpoint_url)
+
+
+def get_s3_path_parts(s3url) -> Tuple[str, str]:
```

Comment:
```python
def get_s3_path_parts(s3url: str) -> Tuple[str, str]:
```
Also, add a docstring after (and for some of the other functions below without them)?